### PR TITLE
[HEAP-15862] convert log utility to TypeScript

### PR DESCRIPTION
**Pull request by dlo**
https://heapinc.atlassian.net/browse/HEAP-15862

HEAP-15862

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -57,10 +68,10 @@ if (isECS) {
     // Otherwise, we expect the meta object to be tags for the log line. Apply some rudimentary
     // sanitizing here to stringify unexpected nested objects. All of our tags should be either
     // string or number values.
-    for (var key of Object.keys(meta)) {
-      var value = meta[key];
-      var isString = typeof value === 'string';
-      var isNumber = typeof value === 'number';
+    for (const key of Object.keys(meta)) {
```
Comment (jamesnghiem) [inline index.ts:71]: NCE - we could change this to a `forEach` block

File: package.json [hunk 2]
```diff
@@ -7,5 +7,11 @@
     "fclone": "1.0.11",
     "winston-log-and-exit": "1.0.6",
     "yargs": "3.30.0"
+  },
+  "devDependencies": {
+    "@types/node": "10.17.24",
+    "@types/winston": "2.2.0",
+    "@types/yargs": "15.0.5",
+    "winston": "2.2.0"
```
Comment (jamesnghiem) [inline package.json:15]: hmm weird that we didn't already ahve `winston` as a dependency considering it was used before. Is it not under `dependencies`?

**Review by jamesnghiem — APPROVED** [review]
1 question about the winston dependency, otherwise LGTM

File: package.json (hunk 2 above)
Comment (dlo) [reply package.json:15]: I'll check on this; it might have been a transitive dependency.

**Pull request merged**